ibis/expr/operations/udf.py

```diff
@@ -78,33 +78,47 @@ def _make_node(
         input_type: InputType,
         name: str | None = None,
         schema: str | None = None,
+        sig: tuple[tuple, Any] | None = None,
```

Comment:
Not really a huge fan of the aesthetics of `sig`.

I don't like that the entire signature is specified as another keyword argument and I don't like all the punctuation.

Anyway, I'm an old man yelling at a cloud :)

Some alternatives that I like better:

```python
# input types must be a positional argument if passed, `returns` must be
# specified as a keyword argument.
@builtin((int, int), returns=str, **kwargs)
```

```python
# this may be controversial and/or require a bunch of invasive changes to
# implement :)
@builtin.signature(int, int).returns(str).options(**kwargs)
```

I think I could be convinced by `signature` if you don't like any of my alternatives, but at the very least `sig` feels a bit too short.

Reply:
I don't mind the former, but it's tricky to implement in a sane way since the first positional argument may also accept a function. I really don't like the builder api in option 2 :).

Went with renaming to `signature` for now (these APIs are still experimental, we can always change it later).

While a few more keystrokes, this syntax has the perk of expanding nicely to support overloads as `signature=[((int, int), int), ((float, float), float)]` if we ever decide we want to support those. Having two arguments for `input_types` and `returns` would be harder to support this use case.

Reply:
I agree with everything here, but I think the current API is the best option I've seen.